calico_kubernetes/calico_kubernetes.py

```diff
@@ -508,43 +528,35 @@ def _get_pod_ports(self, pod):
         return ports
 
     def _get_pod_config(self):
-        """Get the list of pods from the Kube API server."""
-        pods = self._get_api_path('pods')
-        logger.debug('Got pods %s', pods)
-
-        for pod in pods:
-            logger.debug('Processing pod %s', pod)
-            ns = pod['metadata']['namespace'].replace('/', '_')
-            name = pod['metadata']['name'].replace('/', '_')
-            if ns == self.namespace and name == self.pod_name:
-                this_pod = pod
-                break
-        else:
-            raise KeyError('Pod not found: ' + self.pod_name)
-        logger.debug('Got pod data %s', this_pod)
-        return this_pod
+        """Get the pod resource from the API.
+        API Path depends on the api_root, namespace, and pod_name
 
-    def _get_api_path(self, path):
-        """Get a resource from the API specified API path.
-
-        e.g.
-        _get_api_path('pods')
-
-        :param path: The relative path to an API endpoint.
-        :return: A list of JSON API objects
-        :rtype list
+        :return: JSON object containing the pod spec
         """
-        logger.info('Getting API Resource: %s%s', self.api_root, path)
-        session = requests.Session()
-        if self.auth_token:
-            session.headers.update({'Authorization':
-                                    'Bearer ' + self.auth_token})
-        response = session.get(self.api_root + path, verify=False)
-        response_body = response.text
-
-        # The response body contains some metadata, and the pods themselves
-        # under the 'items' key.
-        return json.loads(response_body)['items']
+        with requests.Session() as session:
+            if self._api_root_secure() and self.auth_token:
+                session.headers.update({'Authorization':
+                                        'Bearer ' + self.auth_token})
+
+            path = os.path.join(self.api_root,
+                                'namespaces/%s/pods/%s' % (self.namespace,
+                                                           self.pod_name))
+            try:
+                logger.debug('Fetching API Resource from %s', path)
```

Comment:
"Fetching API resource from %s" is more generic that what is actually going on.

To be clear, this log should just say "Querying API for Pod: %s"